core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1751,6 +1752,25 @@ private void unparseRegexFunctionsOperands(SqlWriter writer, int leftPrec, int r
       }
     }
   }
+  /**
+   * This method is to unparse the REGEXP_REPLACE operands.
+   */
+  private void unparseRegexpReplaceFunctionOperands(SqlWriter writer, int leftPrec, int rightPrec,
+      int indexOfRegexOperand, List<SqlNode> operandList) {
+    int operandListSize = operandList.size();
+    for (int index = 0; index < operandListSize; index++) {
+      if (index < 3) {
```

Comment:
Please attempt to optimize this code and avoid the use of any type of loop.

Reply:
Done